src/sources/nats.rs

```diff
@@ -0,0 +1,181 @@
+use crate::{
+    config::{log_schema, DataType, SourceConfig, SourceContext, SourceDescription},
+    event::{Event, Value},
+    internal_events::NatsEventReceived,
+    shutdown::ShutdownSignal,
+    Pipeline,
+};
+use bytes::Bytes;
+use chrono::Utc;
+use futures::{pin_mut, stream, SinkExt, Stream, StreamExt};
+use serde::{Deserialize, Serialize};
+use snafu::Snafu;
+
+#[derive(Debug, Snafu)]
+enum BuildError {
+    #[snafu(display("Could not create Nats subscriber: {}", source))]
+    NatsCreateError { source: std::io::Error },
+    #[snafu(display("Could not subscribe to Nats topics: {}", source))]
+    NatsSubscribeError { source: std::io::Error },
+}
+
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
+pub struct NatsSourceConfig {
+    url: String,
+    name: String,
```

Comment:
Should we call this `connection_name` or similar? As is, it seems like it could be confused for the name of the source itself.

Reply:
makes sense, changed name in source and sink